File: app/src/main/java/com/example/aashishcalculator/MainActivity.java
package com.example.aashishcalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioGroup;
import android.widget.Switch;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private EditText costOfServiceText;
    private Button calcButton;
    private RadioGroup radioGroup;

    private Switch roundUpSwitch;

    private TextView tipAmountText;


    private double tipAmount;
    private double tipPercent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        costOfServiceText = findViewById(R.id.editTextTextPersonName);
        calcButton = findViewById(R.id.button);
        radioGroup = findViewById(R.id.radioGroup);
        roundUpSwitch = findViewById(R.id.switch1);
        tipAmountText = findViewById(R.id.textView4);

        calcButton.setOnClickListener(new View.OnClickListener() {
            @Override

            public void onClick(View view) {
                String cost = costOfServiceText.getText().toString();
                int costAmount = Integer.parseInt(cost);
                int checkedRadioButton = radioGroup.getCheckedRadioButtonId();
                if(checkedRadioButton == R.id.radioButton){
                    tipPercent = 0.2;
                }else if(checkedRadioButton == R.id.radioButton2){
                    tipPercent = 0.18;
                }else if(checkedRadioButton == R.id.radioButton3){
                    tipPercent = 0.15;
                }else{
                    tipPercent = 0.0;
                }

                tipAmount = costAmount * tipPercent;
                if(roundUpSwitch.isChecked()){
                    tipAmount = Math.round(tipAmount);
                }
                tipAmountText.setText(""+tipAmount);
            }
        });


    }
}

File: app/src/main/java/com/example/diceroller/MainActivity.kt
package com.example.diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

            val rollButtonMainActivity: Button = findViewById(R.id.button)
            rollButtonMainActivity.setOnClickListener {
                rollDice()
                //val textViewMainActivity: TextView = findViewById(R.id.textViewVM)
                //textViewMainActivity.text = "4"
               // val toastMainActivity = Toast.makeText( this, "Dice Rolled", Toast.LENGTH_LONG)
               //tMainActivity.show()

            }
    }

    private fun rollDice() {
        //create a dice with 6 sides and roll it
        val dice = Dice(6)
        val diceRoll = dice.rollMA()

        //update the screen with dice roll
        val resultTextView: TextView = findViewById(R.id.textViewVM)
        resultTextView.text = diceRoll.toString()
    }


        class Dice( val numSides: Int) {
            fun rollMA(): Int {
                return (1..numSides).random()
            }
        }
}